database/gormdb.go

```diff
@@ -42,12 +43,15 @@ type GormDB struct {
 // NewGormDB creates a new gorm database using the provided driver.
 func NewGormDB(path string, logger *zap.Logger) (*GormDB, error) {
 	conn, err := gorm.Open(sqlite.Open(path), &gorm.Config{
-		Logger: NewGORMLogger(logger),
+		Logger:                 NewGORMLogger(logger),
+		SkipDefaultTransaction: true,
```

Comment:
Having this set to `true` may lead to some unexpected results. 
This is especially true for nested types, e.g `Submission{Scores:{}}`. When creating a submission, the submission will be created successfully even though an error occurs when creating the scores. The current behaviour is that the entire creation is rolled back if at any point an error occurs. The same goes for deleting and updating.